### PR TITLE
Fix(Core/Stratholme): Some adjustments to Jarien and Sothos fight

### LINKED ISSUE #9865: Loot of Sothos and Jarien's Heirlooms chest is available to pick up regardless of `Loot: Type`
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

Original report: https://github.com/chromiecraft/chromiecraft/issues/2650

Sothos and Jarien dropping chest that drops item that is not rollable but lootable, so whoever loots chest can pick item without rolling

Tester:

The issue is confirmed. Loot of [Sothos and Jarien's Heirlooms](https://wowgaming.altervista.org/aowow/?object=181083) chest is available to pick up regardless of `Loot: Type`

https://user-images.githubusercontent.com/85256266/147284402-279621e9-2c61-40c3-9c32-8127aa2c9867.mp4

### Expected Blizzlike Behaviour

Chest item should be rollable between people in group

### Source

_No response_

### Steps to reproduce the problem

.go c id 10812 - teleport to Balthazar
.cast 27191
kill Sothos and Jarien
loot Heirlooms

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/c70cc855ff3d2724a0e0a3866b6a43db1f2faddb

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### LINKED ISSUE #10258: Several issues with Jarien and Sothos fight
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

**Issue 1 - Jarien enrage message**

Original report: https://github.com/chromiecraft/chromiecraft/issues/2830

Text reads as "Jarien goes into a rage after seeing $n fall in battle".

WoWScrnShot_011822_204129

Tester:

The issue is confirmed.

WoWScrnShot_012022_104814

**Issue 2 - Jarien and Sothos aggroing pet before their conversation ends**

Original report: https://github.com/chromiecraft/chromiecraft/issues/2829

Jarien and Sothos, when summoned, aggroed Warlock's Pet, a Felhunter, before they finished their conversation and aggroed the players.

Tester:

The issue is confirmed.

WoWScrnShot_012022_104724

### Expected Blizzlike Behaviour

Issue 1: Text should read as "Jarien goes into a rage after seeing Sothos fall in battle!"

Issue 2: Jarien and Sothos should only start the fight after they finished their conversation.

### Source

"Quotes" section of this article
https://wowpedia.fandom.com/wiki/Jarien_and_Sothos

### Steps to reproduce the problem

* `.go creature 52149`
* `.additem 6265`
* summon Void Walker
* `.additem 22057`
* use [Brazier of Invocation]
* Jarien and Sothos start attacking Void Walker before conversation ends
* kill Sothos
* watch Jariens enrage message

### Extra Notes

_No response_

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/c1ebda66bae939e7151fc9f43176dd897e8c01b7

### Operating system

Windows 7

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by Nefertumm**
## Changes Proposed:
-  Move ReactState to constructor to avoid aggro issues with pets
-  Added mob reference to the talk method to use $n
-  Fixed group loot issues with heirlooms

## Issues Addressed:

- Closes #10258
- Closes #9865 
- Closes https://github.com/chromiecraft/chromiecraft/issues/2830
- Closes https://github.com/chromiecraft/chromiecraft/issues/2829
- Closes https://github.com/chromiecraft/chromiecraft/issues/2650

## Tests Performed:

- Tested in-game

## How to Test the Changes:

With 2 chars in party:
Set the loot mode as group loot
.go creature 52149
.additem 6265
summon Void Walker
.additem 22057
use [Brazier of Invocation]
Make sure Jarien and Sothos don't attack Void Walker before conversation ends
kill Sothos/Jarien
watch the other one enrage message
Kill the other one
Loot the chest and you can now roll for the items

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by temperrr** [comment]
Tested

- [x] Group loot and roll
- [x] Doesnt attack pet during rp
- [ ] you can attack them during the RP, this is not correct (https://youtu.be/OH9ueV3hP1Q?t=91)

File: src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp [hunk 1]
```diff
@@ -110,6 +110,9 @@ struct boss_jarien : public BossAI
                 return !me->HasUnitState(UNIT_STATE_CASTING);
             });
 
+        me->SetReactState(REACT_PASSIVE);
+        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_NPC);
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp:115]:
```suggestion
        me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE|UNIT_FLAG_IMMUNE_TO_NPC);
```

**Review by Nyeriah — CHANGES_REQUESTED** [review]

File: src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp (hunk 1 above)
Comment (Nefertumm) [reply src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp:115]: Yes! Good catch!

Comment (Nefertumm) [reply src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp]: Same with remove flag right?

File: src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp [hunk 2]
```diff
@@ -230,6 +233,9 @@ struct boss_sothos : public BossAI
...
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Stratholme/boss_jarien_and_sothos.cpp:238]: here too

**Review by Nyeriah — CHANGES_REQUESTED** [review]

**Comment by Nefertumm** [comment]
Last time 4Head

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**